src/Twig/StimulusTwigExtension.php

```diff
@@ -73,6 +73,79 @@ public function renderStimulusController(Environment $env, $dataOrControllerName
         return rtrim('data-controller="'.implode(' ', $controllers).'" '.implode(' ', $values));
     }
 
+    /**
+     * @param string|array $dataOrControllerName This can either be a map of controller names
+     *                                           as keys set to their "actions" and "events".
+     *                                           Or this can be a string controller name and
+     *                                           action and event are passed as the 2nd and 3rd arguments.
+     * @param string|null  $actionName The action to trigger if a string is passed to the 1st argument. Optional.
+     * @param string|null  $eventName  The event to listen to trigger if a string is passed to the 1st argument. Optional.
+     *
+     * @return string
+     * @throws \Twig\Error\RuntimeError
+     */
+    public function renderStimulusAction(Environment $env, $dataOrControllerName, string $actionName = null, string $eventName = null): string
```

Comment:
Given that we're already documenting it as such in the doc block, does it make sense to use a union type for `$dataOrControllerName`? This would apply to the `renderStimulusAction` method as well.
```suggestion
    public function renderStimulusAction(Environment $env, string|array $dataOrControllerName, string $actionName = null, string $eventName = null): string
```

Reply:
👎 for me, this syntax is only compatible from PHP 8 (where the bundle also requires to be compatible with PHP 7.1.3+)

Reply:
that's a very good point, agreed!